packages/gestalt/src/Button.js

```diff
@@ -136,214 +106,165 @@ function InternalButtonContent({
+      iconEnd,
+      onClick,
+      tabIndex = 0,
+      selected = false,
+      size = 'md',
+      text,
+      type,
+    } = props;
+
+    const innerRef = useRef<null | HTMLAnchorElement | HTMLButtonElement>(null);
+
+    // When using both forwardRef and innerRef, React.useimperativehandle() allows a parent component
+    // that renders <Button ref={inputRef} /> to call inputRef.current.focus()
+    useImperativeHandle(ref, () => innerRef.current);
+
+    const {
+      compressStyle,
+      isTapping,
+      handleBlur,
+      handleMouseDown,
+      handleMouseUp,
+      handleTouchStart,
+      handleTouchMove,
+      handleTouchCancel,
+      handleTouchEnd,
+    } = useTapFeedback({
+      height: innerRef?.current?.clientHeight,
+      width: innerRef?.current?.clientWidth,
+    });
+
+    const { name: colorSchemeName } = useColorScheme();
+    // We need to make a few exceptions for accessibility reasons in darkMode for red buttons
+    const isDarkMode = colorSchemeName === 'darkMode';
+    const isDarkModeRed = isDarkMode && color === 'red';
+    const isDarkModeBlue = isDarkMode && color === 'blue';
+
+    let colorClass = color === 'transparentWhiteText' ? 'transparent' : color;
+
+    if (isDarkModeRed) {
+      colorClass = 'darkModeRed';
+    }
+
+    const { isFocusVisible } = useFocusVisible();
+
+    const sharedTypeClasses = classnames(styles.button, focusStyles.hideOutline, {
+      [styles.inline]: !fullWidth,
+      [styles.block]: fullWidth,
+      [focusStyles.accessibilityOutline]: !disabled && isFocusVisible,
+    });
+
+    const baseTypeClasses = classnames(sharedTypeClasses, touchableStyles.tapTransition, {
+      [styles.sm]: size === 'sm',
+      [styles.md]: size === 'md',
+      [styles.lg]: size === 'lg',
+      // $FlowFixMe[invalid-computed-prop]
+      [styles[colorClass]]: !disabled && !selected,
+      [styles.selected]: !disabled && selected,
+      [styles.disabled]: disabled,
+      [styles.enabled]: !disabled,
+      [touchableStyles.tapCompress]: !disabled && isTapping,
+    });
+
+    const parentButtonClasses = classnames(sharedTypeClasses, styles.parentButton);
+
+    const childrenDivClasses = classnames(baseTypeClasses, styles.childrenDiv);
+
+    const textColor =
+      (disabled && 'subtle') ||
+      (selected && 'inverse') ||
+      ((isDarkModeRed || isDarkModeBlue) && 'default') ||
+      DEFAULT_TEXT_COLORS[color];
+
+    const buttonText = (
+      <Text align="center" color={textColor} overflow="normal" weight="bold">
+        {text}
+      </Text>
     );
-  }
 
-  if (props.type === 'submit') {
-    const { name } = props;
+    if (type === 'submit') {
+      const { name } = props;
+
+      return (
+        <button
+          aria-label={accessibilityLabel}
+          className={baseTypeClasses}
+          data-test-id={dataTestId}
+          disabled={disabled}
+          name={name}
+          onBlur={handleBlur}
+          onClick={(event) => onClick?.({ event })}
```

Comment:
onClick={(event) => onClick?.({ event })}

same as

`onClick={onClick}
`

Reply:
can we merge into a single button?

Reply:
DONE